middleware.ts

```diff
@@ -1,12 +1,10 @@
-import { createMiddlewareClient } from '@supabase/auth-helpers-nextjs'
-import { NextResponse } from 'next/server'
+import { type NextRequest } from 'next/server';
+import { createClient } from '@/utils/supabase/middleware';
 
-import type { NextRequest } from 'next/server'
-import type { Database } from '@/types_db'
-
-export async function middleware(req: NextRequest) {
-  const res = NextResponse.next()
-  const supabase = createMiddlewareClient<Database>({ req, res })
-  await supabase.auth.getSession()
-  return res
-}
\ No newline at end of file
+export async function middleware(request: NextRequest) {
+  const { supabase, response } = createClient(request);
+  // Refresh session if expired - required for Server Components
+  // https://supabase.com/docs/guides/auth/auth-helpers/nextjs#managing-session-with-middleware
+  await supabase.auth.getSession();
+  return response;
+}
```

Comment:
```suggestion
}

export const config = {
  matcher: [
    /*
     * Match all request paths except for the ones starting with:
     * - _next/static (static files)
     * - _next/image (image optimization files)
     * - favicon.ico (favicon file)
     * Feel free to modify this pattern to include more paths.
     */
    '/((?!_next/static|_next/image|favicon.ico).*)',
  ],
}
```

Reply:
Please add this matcher to stop refresh logic in middleware being run for every asset on every load